src/corehost/cli/coreclr.cpp

```diff
@@ -102,10 +139,140 @@ pal::hresult_t coreclr::execute_assembly(
     assert(g_coreclr != nullptr && coreclr_execute_assembly != nullptr);
 
     return coreclr_execute_assembly(
-        host_handle,
-        domain_id,
+        _host_handle,
+        _domain_id,
         argc,
         argv,
         managed_assembly_path,
         exit_code);
 }
+
+pal::hresult_t coreclr_t::create_delegate(
+    const char* entryPointAssemblyName,
+    const char* entryPointTypeName,
+    const char* entryPointMethodName,
+    void** delegate)
+{
+    assert(g_coreclr != nullptr && coreclr_execute_assembly != nullptr);
+
+    return coreclr_create_delegate(
+        _host_handle,
+        _domain_id,
+        entryPointAssemblyName,
+        entryPointTypeName,
+        entryPointMethodName,
+        delegate);
+}
+
+pal::hresult_t coreclr_t::shutdown(int* latchedExitCode)
+{
+    assert(g_coreclr != nullptr && coreclr_shutdown != nullptr);
+
+    bool is_false = false;
+
+    // If already shut down return success since the result
+    // has already been reported to a previous caller.
+    if (!_is_shutdown.compare_exchange_strong(is_false, true))
+    {
+        if (latchedExitCode != nullptr)
+            *latchedExitCode = StatusCode::Success;
+
+        return StatusCode::Success;
+    }
+
+    return coreclr_shutdown(_host_handle, _domain_id, latchedExitCode);
+}
+
+namespace
+{
+    const char *PropertyNameMapping[] =
+    {
```

Comment:
What is this needed for?  In general having this defined in as few places as possible is desirable.

Reply:
Answering my own question - this is a refactor.  Follow-up, any reason for this set and the enum to be in separate files?

Reply:
Yes. There are common values that are always included and there are extra ones. Reconciling them is a goal, but until we have a definitive process for that or what are exposed, I did this. More than willing to put all strings in a single place if that is preferred but operating on enum's is much less error prone than copying around strings. Enums are also much easier to make into stable APIs than strings.